### PR TITLE
[mock-service-host] fix lro status

### PR DESCRIPTION
**Pull request by changlong-liu**
1. fix a bug for failed to set `status: Succeeded` for lro-callback GET.
2.  find resource type with pattern"/providers/microsoft.xxx", (originally with pattern "microsoft.xxx" )

Part of efforts to fixes https://github.com/Azure/autorest.csharp/issues/2226

### CODE REVIEW

File: tools/mock-service-host/src/mid/oav/swaggerMocker.ts [hunk 1]
```diff
@@ -55,7 +55,12 @@ export default class SwaggerMocker {
         let resourceType = ''
         let foundProvider = false
         for (let i = 0; i < pathElements.length; i++) {
-            if (i % 2 === 0 && pathElements[i].match(/microsoft\..+/i)) {
+            if (
+                i > 0 &&
```
Comment (lirenhe) [inline tools/mock-service-host/src/mid/oav/swaggerMocker.ts:59]: You may check whether we should check the resourceprovider name with desc order.  Some paths may contain multiple provider names.

File: tools/mock-service-host/src/mid/oav/swaggerMocker.ts [hunk 2]
```diff
@@ -125,11 +130,16 @@ export default class SwaggerMocker {
...
+                    inUserAssignedIdentities &&
+                    !key.match(
+                        /\/subscriptions\/.*\/providers\/Microsoft.ManagedIdentity\/userAssignedIdentities\/.*/i
+                    )
+                ) {
                     const subscription =
                         pathElements.subscriptions || '00000000-0000-0000-0000-000000000000'
                     const resourceGroup = pathElements.subscriptions || 'mockGroup'
-                    key = `/subscriptions/${subscription}/resourceGroups/${resourceGroup}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/${key}`
+                    key = `/subscriptions/${subscription}/resourceGroups/${resourceGroup}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/aa`
```
Comment (lirenhe) [inline tools/mock-service-host/src/mid/oav/swaggerMocker.ts:142]: why do we need 'aa' for the identity value?

File: tools/mock-service-host/src/mid/oav/swaggerMocker.ts (hunk 1 above)
Comment (changlong-liu) [reply tools/mock-service-host/src/mid/oav/swaggerMocker.ts]: Hi @lirenhe, added logic to use the last provider.

File: tools/mock-service-host/src/mid/oav/swaggerMocker.ts (hunk 2 above)
Comment (changlong-liu) [reply tools/mock-service-host/src/mid/oav/swaggerMocker.ts]: I think here we just want to mock a valid userAssignedIdentity, I changed "aa" to "mocked" to make it more readable. 
The original behavior trapped if there is special characters in `${key}`

**Review by lirenhe — APPROVED** [review]
LGTM

File: tools/mock-service-host/src/mid/responser.ts [hunk 3]
```diff
@@ -355,6 +355,9 @@ export class ResponseGenerator {
             }
             this.swaggerMocker.patchExampleResponses(example, liveRequest)
         }
+        if (liveRequest.query?.[LRO_CALLBACK] === 'true') {
+            this.setStatusToSuccess(example.responses[statusCode].body)
+        }
```
Comment (tadelesh) [inline tools/mock-service-host/src/mid/responser.ts:360]: Which situation will fall into this condition but not be handled by the following logic?

Comment (tadelesh) [reply tools/mock-service-host/src/mid/responser.ts:360]: Discussed offline.

**Review by tadelesh — APPROVED** [review]
LGTM

**Pull request merged**